File: jrc_abogados.client/src/app/Models/Cliente.ts
import { Ubicacion } from "./Ubicacion";

export class Cliente {
  id: number = 0;
  nombre: string = "";
  apellido: string = "";
  fechaNacimiento: any = "";
  telefono = "";
  correoElectronico: string = "";
  ubicacionId: number = 0;
  ubicacion: Ubicacion = new Ubicacion;
  empleadoId: number = 0;
  empleado: any;
  [key: string]: any;
}

File: JRC_Abogados.Server/Models/Cliente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JRC_Abogados.Server.Models
{
    public class Cliente
    {
        [Key]
        public int Id { get; set; }
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public DateTime FechaNacimiento { get; set; }
        public string Telefono { get; set; }
        public string CorreoElectronico { get; set; }
        public int UbicacionId { get; set; }
        [ForeignKey("UbicacionId")]
        public virtual Ubicacion? Ubicacion { get; set; }
        public int EmpleadoId { get; set; }
        [ForeignKey("EmpleadoId")]
        public virtual Empleado? Empleado { get; set; }
    }
}
